pennylane/ops/functions/equal.py

```diff
@@ -482,3 +483,25 @@ def _equal_shadow_measurements(op1: ShadowExpvalMP, op2: ShadowExpvalMP, **_):
 @_equal.register
 def _equal_counts(op1: CountsMP, op2: CountsMP, **kwargs):
     return _equal_measurements(op1, op2, **kwargs) and op1.all_outcomes == op2.all_outcomes
+
+
+@_equal.register
+# pylint: disable=unused-argument
+def _equal_basis_rotation(
+    op1: qml.BasisRotation,
+    op2: qml.BasisRotation,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    if not qml.math.allclose(
+        op1.hyperparameters["unitary_matrix"],
+        op2.hyperparameters["unitary_matrix"],
+        atol=atol,
+        rtol=rtol,
+    ):
+        return False
```

Comment:
While the "unitary matrix" isn't really "trainable", it can still belong to any of the interfaces.  Because of this, let's also add the `check_interface` check.

See https://github.com/PennyLaneAI/pennylane/blob/49b0b71f873e5a6e61ef14472eb99de0af0bab5f/pennylane/ops/functions/equal.py#L259

for an example of how to do it.